src/search/landmarks/landmark_factory_rpg_exhaust.cc

```diff
@@ -27,20 +28,21 @@ void LandmarkFactoryRpgExhaust::generate_relaxed_landmarks(
 
     // insert goal landmarks and mark them as goals
     for (FactProxy goal : task_proxy.get_goals()) {
-        LandmarkNode &lmp = lm_graph->add_simple_landmark(goal.get_pair());
-        lmp.is_true_in_goal = true;
+        vector<FactPair> fact = {goal.get_pair()};
+        Landmark landmark(fact, false, false, true);
+        lm_graph->add_landmark(move(landmark));
     }
     // test all other possible facts
     State initial_state = task_proxy.get_initial_state();
     for (VariableProxy var : task_proxy.get_variables()) {
         for (int value = 0; value < var.get_domain_size(); ++value) {
             const FactPair lm(var.get_id(), value);
             if (!lm_graph->contains_simple_landmark(lm)) {
-                vector<FactPair> facts = {lm};
-                LandmarkNode node(facts, false, false);
+                vector<FactPair> fact = {lm};
```

Comment:
I don't think `lm` is needed outside the if. I'd define it in-place:
`vector<FactPair> fact = {FactPair(var.get_id(), value)};`

Reply:
I also do not like the names `lm` (because it implies the type is `Landmark`) and `fact` (I'd argue that it should be `facts` and the point that there is only one entry should not be reflected in the name).

Reply:
> I don't think `lm` is needed outside the if. I'd define it in-place:
> `vector<FactPair> fact = {FactPair(var.get_id(), value)};`

```lm``` is used inside the ```if```

Reply:
> I also do not like the names `lm` (because it implies the type is `Landmark`) and `fact` (I'd argue that it should be `facts` and the point that there is only one entry should not be reflected in the name).

We called it fact for simple landmarks and facts for disjunctive and conjunctive ones. I don't have a strong opinion for either naming schema so I change it.

Reply:
We changed the constructor to take a `const vector<FactPair>` in order to allow generating the fact in-place.